core/cmd/local_client.go

```diff
@@ -153,72 +153,102 @@ func initLocalSubCmds(client *Client, devMode bool) []cli.Command {
-					Flags:  []cli.Flag{},
-				},
-				{
-					Name:   "status",
-					Usage:  "Display the current database migration status.",
-					Action: client.StatusDatabase,
-					Flags:  []cli.Flag{},
-				},
-				{
-					Name:   "migrate",
-					Usage:  "Migrate the database to the latest version.",
-					Action: client.MigrateDatabase,
-					Flags:  []cli.Flag{},
-				},
-				{
-					Name:   "rollback",
-					Usage:  "Roll back the database to a previous <version>. Rolls back a single migration if no version specified.",
-					Action: client.RollbackDatabase,
-					Flags:  []cli.Flag{},
+			Subcommands: initDbCmds(client, devMode),
+		},
+	}
+}
+
+func initDbCmds(client *Client, devMode bool) []cli.Command {
+	cmds := initPublicDBCmds(client)
+	return append(cmds, initHiddenDBCmds(client, devMode)...)
+}
+
+// initHiddenDBCmds initializes development-mode subcommands
+// hidden db commands skips configuration validation
+// this enables development without a full set of valid secrets, since config validation includes secret validation.
+func initHiddenDBCmds(client *Client, devMode bool) []cli.Command {
+	return []cli.Command{{
+		Name:   "reset",
+		Usage:  "Drop, create and migrate database. Useful for setting up the database in order to run tests or resetting the dev database. WARNING: This will ERASE ALL DATA for the specified database, referred to by CL_DATABASE_URL env variable or by the Database.URL field in a secrets TOML config.",
+		Hidden: !devMode,
+		Action: client.ResetDatabase,
+		Flags: []cli.Flag{
+			cli.BoolFlag{
+				Name:  "dangerWillRobinson",
+				Usage: "set to true to enable dropping non-test databases",
+			},
+		},
+	},
+		{
+			Name:   "preparetest",
+			Usage:  "Reset database and load fixtures.",
+			Hidden: !devMode,
+			Action: client.PrepareTestDatabase,
+			Flags: []cli.Flag{
+				cli.BoolFlag{
+					Name:  "user-only",
+					Usage: "only include test user fixture",
 				},
-				{
-					Name:   "create-migration",
-					Usage:  "Create a new migration.",
-					Hidden: !devMode,
-					Action: client.CreateMigration,
-					Flags: []cli.Flag{
-						cli.StringFlag{
-							Name:  "type",
-							Usage: "set to `go` to generate a .go migration (instead of .sql)",
-						},
-					},
+			},
+		},
+		{
+			Name:   "create-migration",
+			Usage:  "Create a new migration.",
+			Hidden: !devMode,
+			Action: client.CreateMigration,
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  "type",
+					Usage: "set to `go` to generate a .go migration (instead of .sql)",
 				},
 			},
 		},
 	}
 }
 
+// initPublicDBCmds initializes public facing commands.
+// public db commands have config validation applied to them
+func initPublicDBCmds(client *Client) []cli.Command {
+	return []cli.Command{
+		{
+			Name:   "version",
+			Usage:  "Display the current database version.",
+			Action: client.VersionDatabase,
+			Flags:  []cli.Flag{},
+			Before: func(ctx *clipkg.Context) error {
+				return client.Config.Validate()
```

Comment:
I think it's important to do minimal validation if possible.
I had imagined that we could add a new method:
```suggestion
				return client.Config.ValidateDatabase()
```
Or combine it with a getter for the subset of methods used, so it's less brittle, e.g.:
```suggestion
				client.DatabaseConfig, err = client.Config.Database()
				return err
```
Does this seem like a viable approach?

Reply:
could probably makes something like that work. do the db secrets need to be validated as part of this? is the toml validation only with regard to the [[Database]] or whatever the associated tag is?

Reply:
I had initially thought we could just validate whichever fields are being used, and hopefully leverage existing methods like: https://github.com/smartcontractkit/chainlink/blob/489280eb6530092afe8711ca059d201d7091e97e/core/config/v2/types.go#L132
I'm second guessing now since some things only seem critical when actually running a node, but we do still want those standard 'missing' errors, so probably still good to start with these and see if it works :shrug: 
Do we use any fields outside of the two `[Database]` tables?

Reply:
i read the db code more closely and i'm confused about the goal of the ticket. i thought there was already validation logic written, ie the config validation, that needed to be applied to this set of commands. that was the wrong interpretation. 


what does it mean to validate the db config? i know you have said it out loud to me and i thought i understood but i don't. the more specific the better and i would love some examples.

all the functions have good error handling for the db url and i don't see much else in common to validate.

that it, it seems like the only thing you could hope to validate is the form of the dburl or perhaps the existence of some other things, perhaps the secrets, but if all the commands connect to the db and we have good error handling, then what does upfront validation buy us?

Reply:
> i thought there was already validation logic written, ie the config validation, that needed to be applied to this set of commands. that was the wrong interpretation.

That was the plan, but only a subset of the validation.

> what does it mean to validate the db config?

By 'db config' I just mean whichever fields are being used by this set of commands. I suspected that there may be other random global/root level config fields used by these commands.

> all the functions have good error handling for the db url and i don't see much else in common to validate.
>
> that it, it seems like the only thing you could hope to validate is the form of the dburl or perhaps the existence of some other things, perhaps the secrets, but if all the commands connect to the db and we have good error handling, then what does upfront validation buy us?

That's great! In that case there may not be any outstanding issues per se, but it would still be beneficial to share the validation code one way or another to align the UX, since the new TOML validation introduces a standard error format, and to keep things coupled going forward.

Reply:
can you point me to the standard error format that you mentioned? i can look at aligning that reporting

Reply:
The `ValidateConfig` impls return standard multi-errors:
https://github.com/smartcontractkit/chainlink/blob/455f908fd9dcb2a4bb8f313578361dfeca10fa60/core/config/v2/validate.go#L14-L21
which are formatted by a utils func:
https://github.com/smartcontractkit/chainlink/blob/455f908fd9dcb2a4bb8f313578361dfeca10fa60/core/config/v2/validate.go#L26

Reply:
i've redone this all once more. i think i've finally got it.